.github/workflows/docker-publish.yml

```diff
@@ -94,3 +94,74 @@ jobs:
         # This step uses the identity token to provision an ephemeral certificate
         # against the sigstore community Fulcio instance.
         run: echo "${TAGS}" | xargs -I {} cosign sign --yes {}@${DIGEST}
+
+
+  # build-glr:
+  #   needs: build
+  #   runs-on: ubuntu-latest
+  #   permissions:
+  #     contents: read
+  #     packages: write
+  #     # This is used to complete the identity challenge
+  #     # with sigstore/fulcio when running outside of PRs.
+  #     id-token: write
+  #   steps:
+  #     - name: Checkout repository
+  #       uses: actions/checkout@v4
+
+  #     # Install the cosign tool except on PR
+  #     # https://github.com/sigstore/cosign-installer
+  #     - name: Install cosign
+  #       if: github.event_name != 'pull_request'
+  #       uses: sigstore/cosign-installer@59acb6260d9c0ba8f4a2f9d9b48431a222b68e20 #v3.5.0
+  #       with:
+  #         cosign-release: 'v2.2.4'
+
+  #     # Set up BuildKit Docker container builder to be able to build
+  #     # multi-platform images and export cache
+  #     # https://github.com/docker/setup-buildx-action
+  #     - name: Set up Docker Buildx
+  #       uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+
+  #     # Login against a Docker registry except on PR
+  #     # https://github.com/docker/login-action
+  #     - name: Log into registry ${{ env.REGISTRY }}
+  #       if: github.event_name != 'pull_request'
+  #       uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+  #       with:
+  #         registry: ${{ env.REGISTRY }}
+  #         username: ${{ github.actor }}
+  #         password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Docker metadata - glr variant
+        id: meta-glr
+        uses: docker/metadata-action@96383f45573cb7f253c731d3b3ab81c87ef81934 # v5.0.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          flavor: latest=false
+          tags: |
+            type=raw,value=${{ steps.today.outputs.date }}-glr,priority=840
+
+      - name: Build and push Docker image - glr variant
+        id: build-and-push-glr
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
+        with:
+          context: .
+          file: Dockerfile.glr
+          build-args: BASE_VERSION=${{ github.event_name != 'pull_request' && steps.today.outputs.date || 'latest' }}
```

Comment:
So - this isn't great, and I'm hoping someone has an idea about how to do it better.

Ideally - the `-glr` image that's build and released here uses the "standard" image that was build a couple of steps previously, but using the `YYYYMMDD` tag here errors out, at least in the PR case, because the standard date tag hasn't been pushed yet (I guess, I was hoping doing it in one job would let the builder find the local copy that wasn't pushed, but apparently not)

This will hopefully have the correct behavior on merge, which is more important, but it does mean that the "build" test is potentially brittle and meaningless.

Reply:
Do we know another tag that has been pushed at this point? (Ideally `github.sha` so we don't have multiple things potentially stomping each other) Adding the YYMMDD at time of merge seems a better bet.

Reply:
figured out how to do this without publishing the image (hint, turn off `buildx`)